### PR TITLE
Add new endpoints to reset specific value son search indexes

### PR DESCRIPTION
**Pull request by lrojas94**

### CODE REVIEW

File: src/pages/api/mod/search/articles-update.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import { dbWrite } from '~/server/db/client';
+import { z } from 'zod';
+import { ModEndpoint } from '~/server/utils/endpoint-helpers';
+import { MODELS_SEARCH_INDEX } from '../../../../server/common/constants';
+import { updateDocs } from '../../../../server/meilisearch/client';
+import { withRetries } from '../../../../server/utils/errorHandling';
+import { userWithCosmeticsSelect } from '../../../../server/selectors/user.selector';
+
+const BATCH_SIZE = 10000;
+const INDEX_ID = MODELS_SEARCH_INDEX;
```
Comment (manuelurenah) [inline src/pages/api/mod/search/articles-update.ts:11]:
```suggestion
const INDEX_ID = ARTICLES_SEARCH_INDEX;
```

File: src/pages/api/mod/search/articles-update.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    return updateIndexReadyRecords[updateIndexReadyRecords.length - 1].id;
+  });
+
+export default ModEndpoint(
+  async function updateArticlesSearchIndex(req: NextApiRequest, res: NextApiResponse) {
+    const { update } = schema.parse(req.query);
+    const start = Date.now();
+    const updateMethod: ((idOffset: number) => Promise<number>) | null =
+      update === 'user' ? updateUser : null;
```
Comment (manuelurenah) [inline src/pages/api/mod/search/articles-update.ts:70]: Does it makes sense to condition the updateMethod? Are we going to support others as well?
```suggestion
    const updateMethod: ((idOffset: number) => Promise<number>) | null =
      update === 'article' ? updateUser : null;
```

File: src/pages/api/mod/search/images-update.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  Prisma.sql`i."ingestion" = ${ImageIngestionStatus.Scanned}::"ImageIngestionStatus"`,
+  Prisma.sql`i."tosViolation" = false`,
+  Prisma.sql`i."type" = 'image'`,
+  Prisma.sql`i."needsReview" IS NULL`,
+  Prisma.sql`p."publishedAt" IS NOT NULL`,
+  Prisma.sql`p."availability" != 'Private'::"Availability"`,
+];
+
+const schema = z.object({
+  update: z.enum(['user']),
```
Comment (manuelurenah) [inline src/pages/api/mod/search/images-update.ts:29]: Same
```suggestion
  update: z.enum(['images']),
```

File: src/pages/api/mod/search/images-update.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+});
+
+const updateUserDetails = (idOffset: number) =>
```
Comment (manuelurenah) [inline src/pages/api/mod/search/images-update.ts:32]:
```suggestion
const updateImage = (idOffset: number) =>
```

File: src/pages/api/mod/search/images-update.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+      batchSize: BATCH_SIZE,
+    });
+
+    console.log('Indexed records count: ', updateIndexReadyRecords.length);
+
+    return updateIndexReadyRecords[updateIndexReadyRecords.length - 1].id;
+  });
+
+export default ModEndpoint(
+  async function updateModelSearchIndex(req: NextApiRequest, res: NextApiResponse) {
```
Comment (manuelurenah) [inline src/pages/api/mod/search/images-update.ts:154]:
```suggestion
  async function updateImageSearchIndex(req: NextApiRequest, res: NextApiResponse) {
```

File: src/pages/api/mod/search/images-update.ts [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    const { update } = schema.parse(req.query);
+    const start = Date.now();
+    const updateMethod: ((idOffset: number) => Promise<number>) | null =
+      update === 'user' ? updateUserDetails : null;
```
Comment (manuelurenah) [inline src/pages/api/mod/search/images-update.ts:158]:
```suggestion
    const updateMethod: ((idOffset: number) => Promise<number>) | null =
      update === 'image' ? updateUserDetails : null;
```

File: src/pages/api/mod/search/articles-update.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    not: null,
+  },
+  tosViolation: false,
+});
+
+const schema = z.object({
+  update: z.enum(['user']),
+});
+
+const updateUser = (idOffset: number) =>
```
[the remark this answers is not in the document]
Comment (lrojas94) [reply src/pages/api/mod/search/articles-update.ts:24]: No, it should be `updateUser` The idea is to call each method what it does. This method updates the user within the articles.

File: src/pages/api/mod/search/articles-update.ts (hunk 2 above)
Comment (lrojas94) [reply src/pages/api/mod/search/articles-update.ts:70]: Same as above, `user` is accurate

**Review by manuelurenah — APPROVED** [review]

**Pull request merged**